src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java

```diff
@@ -435,6 +435,59 @@ void should_save_task_and_get_task_from_task_assignments() {
                 });
     }
 
+    @Test
+    void should_save_task_and_get_task_from_task_assignments_with_location_null() {
+        TaskResource taskResource = createAndAssignTask();
+        taskResource.setLocation(null);
+        checkHistory(taskResource.getTaskId(), 1);
+
+        await().ignoreException(AssertionFailedError.class)
+            .pollInterval(1, SECONDS)
+            .atMost(10, SECONDS)
+            .until(
+                () -> {
+                    List<TaskAssignmentsResource> taskAssignmentsList
+                        = miReportingServiceForTest.findByAssignmentsTaskId(taskResource.getTaskId());
+
+                    assertFalse(taskAssignmentsList.isEmpty());
+                    assertEquals(1, taskAssignmentsList.size());
+                    assertEquals(taskResource.getTaskId(), taskAssignmentsList.get(0).getTaskId());
+                    assertEquals(taskResource.getTaskName(), taskAssignmentsList.get(0).getTaskName());
+                    assertEquals(taskResource.getAssignee(), taskAssignmentsList.get(0).getAssignee());
+                    assertEquals(taskResource.getJurisdiction(), taskAssignmentsList.get(0).getService());
+                    assertNull(taskAssignmentsList.get(0).getAssignmentEnd());
+                    assertNull(taskAssignmentsList.get(0).getAssignmentEndReason());
+
+                    return true;
+                });
+    }
+
+    @Test
+    void should_save_task_and_get_task_from_task_assignments_with_taskName_null() {
+        TaskResource taskResource = createAndAssignTask();
+        taskResource.setTaskName(null);
+        checkHistory(taskResource.getTaskId(), 1);
```

Comment:
Same as above comments for task_name too, after setting to null need to save taskResoource. Also need to add an assertion that the taskAssignement is created with task_name as null